### DISCUSSION TITLE
Allow `using` declarations without `var` )

### DISCUSSION THREAD

**Discussion by lucasteles** [dotnet/csharplang#6860] [Language Ideas]
In C#8 we had the [using declarations](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/using.md#using-declaration) feature witch allows

Write this:
```cs
{
    using (var scoped  = logger.BeginScope("..."))
    {
        logger.LogInformation("foo");
        logger.LogInformation("bar");
    }
}
```
as this
```cs
{
    using var scoped = logger.BeginScope("...");
    logger.LogInformation("foo");
    logger.LogInformation("bar");
}
```
Witch simplifies the code reading mainly by removing one indentation level.

But, if I don't care about the scoped variable I have to use the block form

```cs
{
    using (logger.BeginScope("..."))
    {
        logger.LogInformation("foo");
        logger.LogInformation("bar");
    }
}
```

A `workaround` is to use the `discard` symbol, which is not a discard in this context, it is defining the variable `_`, so it is not ideal 
```cs
{
    using var _  = logger.BeginScope("...");
}
```

The **proposal** is just to allow the use of scoped disposable `using` with a discard or alone

```cs
{
    // this
    using logger.BeginScope("...");
    
    // or this
    using  _  = logger.BeginScope("...");
}
```

**Comment by CyrusNajmabadi** [comment]
The issue here is that this would now become ambiguous: `using A.B;`.  Parsing this can legally make a using-directive, *or* a using-statement.  Both of those are legal at the top level of a program.

**Reply by HaloFour** [reply]
That'd also be true for using discards here:

```cs
using _ = System.Console; // legal today, defines an alias
```

**Comment by yaakov-h** [comment]
Using with a discard is a duplicate of #2235.

**Comment by The-Futurist** [comment]
This would not be ambiguous _within a method block_ though which is what is being proposed; it can't (legally) _be_ a using directive so how could it be ambiguous with that?

**Reply by HaloFour** [reply]
It is expected that anything that is legal within a method block is also legal as a top level statement.

**Comment by The-Futurist** [comment]
That's true, I didn't consider that, however if top level statements have led to restrictions like this, one has to wonder - was it worth it? I personally don't see much value in that feature, but some might.

**Reply by CyrusNajmabadi** [reply]
> however if top level statements have led to restrictions like this, one has to wonder - was it worth it?

It's a very minor restriction, vs a large user gain.  Overall, that seems like a great exchange.  :)

**Reply by foxesknow** [reply]
What measurements did you do to reach the conclusion it is a "large user gain" ?
